resource_mailgun_domain.go

```diff
@@ -137,19 +144,25 @@ func resourceMailgunDomainCreate(d *schema.ResourceData, meta interface{}) error
 }
 
 func resourceMailgunDomainDelete(d *schema.ResourceData, meta interface{}) error {
-	client := *meta.(*mailgun.Mailgun)
+	client := *meta.(*mailgun.MailgunImpl)
 
 	log.Printf("[INFO] Deleting Domain: %s", d.Id())
 
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
 	// Destroy the domain
-	err := client.DeleteDomain(d.Id())
+	err := client.DeleteDomain(ctx, d.Id())
 	if err != nil {
 		return fmt.Errorf("Error deleting domain: %s", err)
 	}
 
 	// Give the destroy a chance to take effect
 	return resource.Retry(1*time.Minute, func() *resource.RetryError {
-		_, _, _, err = client.GetSingleDomain(d.Id())
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+		defer cancel()
```

Comment:
30 seconds is quite a long time for a http request to fail, but it seems interesting to hard code this into the client. I don't think I've seen other timeouts hardcoded into terraform clients.

At a minimum can we extract the `time.seconds*30` into a constant and use it here and below?

Reply:
It looks like just `context.Background()` should be used, examples:
- https://github.com/terraform-providers/terraform-provider-google/blob/1319ddd86de649e62984da8a31ca5643c4ef1787/google/resource_google_project_services.go#L209
- https://github.com/terraform-providers/terraform-provider-github/blob/6ed5159cce57eee82a11546135579e0b508adf81/github/resource_github_membership.go#L48

So maybe I just change the code to that?